services/requester/requester.go

```diff
@@ -202,6 +216,13 @@ func (e *EVM) SendRawTransaction(ctx context.Context, data []byte) (common.Hash,
 		return common.Hash{}, fmt.Errorf("failed to derive the sender: %w", err)
 	}
 
+	if e.config.FlowNetworkID == flowGo.Testnet {
```

Comment:
I would suggest to check whether `tx-request-limit > 0`:
```
rateLimitEnabled := tx-request-limit > 0
if rateLimitEnabled {
  ...
}
```

So that by default, the rate limit is disabled, for testnet, we will add a flag `tx-request-limit=1` to enable it.

Reply:
+1. let's make this flexible so it could be used anywhere, and let the operator decide.

Reply:
Good idea :+1: Updated in https://github.com/onflow/flow-evm-gateway/pull/750/commits/7b49b082cca6eb110c34df2cbefeb945a7f81d77